GBShop/GBShop/App/Flows/Screens/SuplimentaryScreens/SignUpViewController.swift

```diff
@@ -0,0 +1,213 @@
+
+            switch response.result {
+            case .success(let model):
+                let resultWithSignUpSuccess: Int = 1
+                #if DEBUG
+                print(model)
+                #endif
+                DispatchQueue.main.async { [weak self] in
+                    let handler: ((UIAlertAction) -> Void)? = { [weak self] _ in self?.dismiss(animated: true, completion: nil)
+                    }
+                    guard model.result == resultWithSignUpSuccess else {
+                        self?.showAlert(
+                            title: NSLocalizedString("signup", comment: ""),
+                            message: NSLocalizedString("signupFailure", comment: ""),
+                            handler: handler,
+                            completion: nil
+                        )
+                        return
+                    }
+                    self?.showAlert(
+                        title: NSLocalizedString("signup", comment: ""),
+                        message: NSLocalizedString("signupSuccess", comment: ""),
+                        handler: handler,
+                        completion: nil
+                    )
+                }
+            case .failure(let error):
+                print(error.localizedDescription)
+                DispatchQueue.main.async { [weak self] in
+                    self?.dismiss(animated: true, completion: nil)
+                }
+            }
+        }
+    }
+
+    @objc func keyboardWillBeShown(notification: Notification) {
+        guard let userInfo = notification.userInfo else {
+            return
+        }
+        let info = userInfo as NSDictionary
+
+        let keyboardSize = (info.value(forKey: UIResponder.keyboardFrameEndUserInfoKey) as? NSValue)?.cgRectValue.size
+        let contentInsets = UIEdgeInsets(
+            top: 0.0,
+            left: 0.0,
+            bottom: keyboardSize?.height ?? 0.0,
+            right: 0.0
+        )
+        signUpView.contentInset = contentInsets
+        signUpView.scrollIndicatorInsets = contentInsets
+    }
+
+    @objc func keyboardWillBeHiden(notification: Notification) {
+        signUpView.contentInset = UIEdgeInsets.zero
+        signUpView.scrollIndicatorInsets = UIEdgeInsets.zero
+    }
+
+    @objc func hideKeyboard() {
+        signUpView.endEditing(true)
+    }
+
+    // MARK: - Private methods
+
+    // MARK: Configure
+
+    private func configureSubviews() {
+        let navigationBarHeight: CGFloat = 56.0
+        let signUpButtonHeight: CGFloat = 56.0
+
+        // Create Navigation Bar with Navigation Item to set the title of the SignUp VC
+
+        let frame = CGRect(
+            x: 0.0,
+            y: 0.0,
+            width: self.view.bounds.size.width,
+            height: navigationBarHeight
+        )
+        let navigationItem = UINavigationItem()
+
+        navigationItem.title = NSLocalizedString("signup", comment: "")
+
+        navigationBar = UINavigationBar(frame: frame)
```

Comment:
А вот это странно. Навбар обычно вручную не создают – он часть UINavigationController и проставляется автоматически для всех дочерних контроллеров.
как оно должно быть: создаешь UINavigationController, в него запихиваешь SignUpViewController – и внутри контроллера делаешь self.title = NSLocalizedString("signup", comment: "").

Reply:
Issue: Create navigation bar with navigation item to set its custom look and the title of the SignIn VC.
Explanation: Because of design, the correct way to show the SignIn VC is to use .present for its navigation controller with SignIn VC modal presentation style as .formSheet instead of just to push it in navigation controller where navigation bar exists by default.